# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/VMFunctions.cpp
# Commit: 4531d5c25694
# Full Hash: 4531d5c256947ff782377ba0143e3f37bd7207c2
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 5 - Remove JSClass* from ObjectGroup. r=jonco
#   
#   BaseShape stores the JSClass* too so this patch changes the code to get the JSClass*
#   from there instead of from the ObjectGroup.
#   
# ==============================================================================

diff -r f1bb16079c0e -r 4531d5c25694 js/src/jit/VMFunctions.cpp
--- a/js/src/jit/VMFunctions.cpp	Fri Mar 05 19:10:06 2021 +0000
+++ b/js/src/jit/VMFunctions.cpp	Fri Mar 05 19:10:06 2021 +0000
@@ -1568,7 +1568,6 @@
   MOZ_ASSERT(obj->compartment() == cx->compartment());
   MOZ_ASSERT(obj->zoneFromAnyThread() == cx->zone());
   MOZ_ASSERT(obj->runtimeFromMainThread() == cx->runtime());
-  MOZ_ASSERT(obj->group()->clasp() == obj->shape()->getObjectClass());
 
   if (obj->isTenured()) {
     MOZ_ASSERT(obj->isAligned());